src/Confluent.Kafka/Consumer.cs

```diff
@@ -645,13 +659,8 @@ public void Assign(IEnumerable<TopicPartition> partitions)
         public void Unassign()
             => kafkaHandle.Assign(null);
 
-        /// <summary>
-        ///     Manually consume message or triggers events.
-        ///
-        ///     Will invoke events for OnPartitionsAssigned/Revoked,
-        ///     OnOffsetCommit, etc. on the calling thread.
-        /// </summary>
-        public bool Consume(out Message message, int millisecondsTimeout)
+
+        internal bool ConsumeImpl(out Message message, int millisecondsTimeout)
```

Comment:
Well, there are many opinions on default parameter values :). They entered the game quite late, so it may be one of the reasons they are not used in BCL. But I guess the most important reason may be potential [compile/runtime time issues](http://haacked.com/archive/2010/08/10/versioning-issues-with-optional-arguments.aspx/). E.g. if you make some param optional or change its default value, the callers may need to be recompiled. As these breaking changes can be quite subtle, IMO they may not be that useful in public APIs...